lib/types/btrfs.nix

```diff
@@ -1,4 +1,52 @@
 { config, options, diskoLib, lib, rootMountPoint, parent, device, ... }:
+let
+  swapType = lib.types.submodule ({ config, ... }: {
+    options = {
+      enable = lib.mkEnableOption "swap";
+
+      size = lib.mkOption {
+        type = lib.types.nullOr (lib.types.strMatching "^([0-9]+[KMGTP])?$");
+        default = null;
+        description = "Size of the swap file (e.g. 2G)";
+      };
```

Comment:
```suggestion
      size = lib.mkOption {
        type = lib.types.strMatching "^([0-9]+[KMGTP])?$";
        description = "Size of the swap file (e.g. 2G)";
      };
```

Reply:
I *think* this might not work. I will test it, but when I was creating this code I just looked at outcomes in `nix repl`, and it would crash on me from `size` not being set on items that did not have `enable = true;`. It might be because I tried to print out values in the repl though. Now that I have a system where I can test this I will see if it works without `null` being a valid value.

Reply:
I've removed the ability to set this to `null` when moving to a list of files, cause the whole issue became moot.